flare/learners/otf.py

```diff
@@ -578,6 +586,25 @@ def run_dft(self):
         f = logging.getLogger(self.output.basename + "log")
         f.info("\nCalling DFT...\n")
 
+        # CP2K calculators cannot be pickled; solution is to just create a new instance
+        if type(self.dft_calc) == CP2K:
+            self.dft_calc.atoms = self.ase_atoms
+            self.dft_calc.write("tmp-cp2k")
+
+            # now re-initiate the CP2K calculator
+            cp2k_command = os.getenv("ASE_CP2K_COMMAND").split()[-1].split(os.sep)[-1]
+            self.atoms.calc = CP2K(restart="tmp-cp2k", command=cp2k_command)
+            self.atoms.calc.calculate(
+                atoms=self.ase_atoms, properties=["forces", "energy", "stress"]
+            )
+            self.atoms.calc = deepcopy(self.ase_atoms)
```

Comment:
Is this correct to have the `calc` assigned an `atoms`?

Reply:
Good catch, this should have said `self.atoms.calc.atoms = deepcopy(self.ase_atoms)` but now the changes are not needed